persistence/block.go

```diff
@@ -55,15 +76,51 @@ func (p PostgresContext) TransactionExists(transactionHash string) (bool, error)
 	return true, err
 }
 
-func (p PostgresContext) StoreTransaction(txResult modules.TxResult) error {
-	return p.txIndexer.Index(txResult)
+func (p PostgresContext) StoreTransactions() error {
+	// TODO: store in batch
+	for _, txResult := range p.GetLatestTxResults() {
+		if err := p.txIndexer.Index(txResult); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (p PostgresContext) StoreBlock(blockProtoBytes []byte) error {
+// DISCUSS: this might be retrieved from the block store - temporarily we will access it directly from the module
+//       following the pattern of the Consensus Module prior to pocket/issue-#315
+func (p *PostgresContext) SetProposalBlock(blockHash string, blockProtoBytes, proposerAddr, qc []byte, transactions [][]byte) error {
+	p.latestBlockHash = blockHash
```

Comment:
Per my comment above, thinking of renaming these to `proposalX` instead of `latestX` since it is only a proposal until the context is commited.

Reply:
See comment above about `ApplyBlock` being used during state sync